mne_bids/mne_bids.py

```diff
@@ -424,60 +459,35 @@ def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
+                  verbose)
     _channels_tsv(raw, channels_fname, verbose)
 
     events = _read_events(events_data, raw)
     if len(events) > 0:
         _events_tsv(events, raw, events_tsv_fname, event_id, verbose)
 
-    # for FIF, we need to re-save the file to fix the file pointer
-    # for files with multiple parts
-    if ext in ['.fif', '.gz']:
-        raw.save(raw_file_bids, overwrite=overwrite)
-    else:
-        if os.path.exists(raw_file_bids):
-            if overwrite:
-                os.remove(raw_file_bids)
-                sh.copyfile(raw_file, raw_file_bids)
-            else:
-                raise ValueError('"%s" already exists. Please set overwrite to'
-                                 ' True.' % raw_file_bids)
-
-    return output_path
-
-
-def _read_events(events_data, raw):
-    """Read in events data.
+    # Writing of neural data
+    # Check if it is MEG (only writing to FIF)
+    # ----------------------------------------
+    if os.path.exists(raw_file_bids) and not overwrite:
+        raise ValueError('"%s" already exists. Please set'
+                         ' overwrite to True.' % raw_file_bids)
 
-    Parameters
-    ----------
-    events_data : str | array | None
-        The events file. If a string, a path to the events file. If an array,
-        the MNE events array (shape n_events, 3). If None, events will be
-        inferred from the stim channel using `find_events`.
-    raw : instance of Raw
-        The data as MNE-Python Raw object.
+    if verbose:
+        print('Writing data files to %s' % raw_file_bids)
 
-    Returns
-    -------
-    events : array, shape = (n_events, 3)
-        The first column contains the event time in samples and the third
-        column contains the event id. The second column is ignored for now but
-        typically contains the value of the trigger channel either immediately
-        before the event or immediately after.
+    if ext in ALLOWED_EXTENSIONS:
+        # for FIF, we need to re-save the file to fix the file pointer
+        # for files with multiple parts
+        if ext in ['.fif', '.gz']:
+            raw.save(raw_file_bids, overwrite=overwrite)
+        else:
+            if os.path.exists(raw_file_bids):  # overwrite=True
+                os.remove(raw_file_bids)
+                sh.copyfile(raw_fname, raw_file_bids)
 
-    """
-    if isinstance(events_data, string_types):
-        events = read_events(events_data).astype(int)
-    elif isinstance(events_data, np.ndarray):
-        if events_data.ndim != 2:
-            raise ValueError('Events must have two dimensions, '
-                             'found %s' % events.ndim)
-        if events_data.shape[1] != 3:
-            raise ValueError('Events must have second dimension of length 3, '
-                             'found %s' % events.shape[1])
-        events = events_data
     else:
-        events = find_events(raw, min_duration=0.001)
-    return events
+        pass
```

Comment:
@sappelhoff I just realized while reviewing https://github.com/mne-tools/mne-bids/pull/69 that this is not a good construct. You are violating two maxims of Zen of Python:

> Flat is better than nested.

and

> Errors should never pass silently. Unless explicitly silenced.